### ISSUE TITLE
Kernel crash with ioclass and seq-cutoff

### ISSUE THREAD

**Issue by inspur-wyq** [Open-CAS/open-cas-linux#988]
## Description

Similar to https://github.com/Open-CAS/open-cas-linux/issues/663, when I set seq-cutoff to 16k and ioclass to 128k, the kernel crashed, but the version is `21.06.1.0547.release` which is after [Don't evict on hit](https://github.com/Open-CAS/ocf/pull/446).
## Steps to Reproduce

1. vgcreate cache-nvme0n1 /dev/nvme0n1
2. lvcreate -l 10%VG -n 1 cache-nvme0n1
3. casadm -S -i 1 -d /dev/disk/by-id/dm-name-cache--nvme0n1-1 -c wb -f
4. vgcreate b-sdc /dev/sdc 
5. lvcreate -l 100%VG -n 1 b-sdc
6. casadm -A -d /dev/disk/by-id/dm-name-b--sdc-1 -i 1 -j 1
7. casadm -X -n cleaning-alru -t 1000 -i 1
8. casadm -X --name seq-cutoff --threshold 16 --policy always -i 1 -j 1
9. casadm --io-class --load-config -i 1 -f ioclass-config.csv
10. cat ioclass-config.csv
```
IO class id,IO class name,Eviction priority,Allocation
0,unclassified,22,0
1,request_size:le:131072,1,1
```
11. fio --filename=/dev/cas1-1 --ioengine=libaio --rw=randwrite --bs=4k --iodepth=64 --numjobs=4 --direct=1 --name=4kiops_randwrite --group_reporting --lockmem=1G --size=4096M
12. fio --filename=/dev/cas1-1 --ioengine=libaio --rw=randread --bs=512K --iodepth=64 --numjobs=1 --direct=1 --name=4kiops_randread --group_reporting --lockmem=1G --size=4096M

## Logs

```
crash> bt
PID: 199874  TASK: ffff8a4c44428000  CPU: 0   COMMAND: "cas_io_13_0"
 #0 [fffffe0000007d58] machine_kexec at ffffffff9846ee83
 #1 [fffffe0000007db8] __crash_kexec at ffffffff98556c82
 #2 [fffffe0000007e88] crash_kexec at ffffffff98557b21
 #3 [fffffe0000007ea8] oops_end at ffffffff984351bd
 #4 [fffffe0000007ed0] die at ffffffff984359a2
 #5 [fffffe0000007f00] handle_stack_overflow at ffffffff98432002
 #6 [fffffe0000007f20] do_double_fault at ffffffff984320f9
 #7 [fffffe0000007f50] double_fault at ffffffff99000d8a
    [exception RIP: ocf_metadata_get_core_info+21]
    RIP: ffffffffc0adc0b5  RSP: ffffa22a0ae84000  RFLAGS: 00010287
    RAX: ffffa22a06725000  RBX: 0000000002e02676  RCX: ffffa22a0ae84020
    RDX: ffffa22a0ae8401a  RSI: 0000000002e02676  RDI: ffffa22a0d743000
    RBP: ffffa22a0ae84008   R8: 0000000002e02676   R9: ffffa22a0d743000
    R10: 0000000000000080  R11: 0000000000000080  R12: ffffa22a0d743000
    R13: 0000000000000001  R14: 00000000000414e0  R15: ffff8a3ccdef4750
    ORIG_RAX: ffffffffffffffff  CS: 0010  SS: 0018
--- <DOUBLEFAULT exception stack> ---
 #8 [ffffa22a0ae84000] ocf_metadata_get_core_info at ffffffffc0adc0b5 [cas_cache]
 #9 [ffffa22a0ae84060] ocf_engine_lookup at ffffffffc0acc2ff [cas_cache]
#10 [ffffa22a0ae840a8] ocf_engine_traverse at ffffffffc0acc492 [cas_cache]
#11 [ffffa22a0ae840c0] ocf_read_pt at ffffffffc0acda34 [cas_cache]
#12 [ffffa22a0ae840e0] ocf_read_generic at ffffffffc0acb6d1 [cas_cache]
#13 [ffffa22a0ae84100] ocf_read_pt at ffffffffc0acdaf5 [cas_cache]
#14 [ffffa22a0ae84120] ocf_read_generic at ffffffffc0acb6d1 [cas_cache]
#15 [ffffa22a0ae84140] ocf_read_pt at ffffffffc0acdaf5 [cas_cache]
#16 [ffffa22a0ae84160] ocf_read_generic at ffffffffc0acb6d1 [cas_cache]
#17 [ffffa22a0ae84180] ocf_read_pt at ffffffffc0acdaf5 [cas_cache]
#18 [ffffa22a0ae841a0] ocf_read_generic at ffffffffc0acb6d1 [cas_cache]
#19 [ffffa22a0ae841c0] ocf_read_pt at ffffffffc0acdaf5 [cas_cache]
#20 [ffffa22a0ae841e0] ocf_read_generic at ffffffffc0acb6d1 [cas_cache]
#21 [ffffa22a0ae84200] ocf_read_pt at ffffffffc0acdaf5 [cas_cache]

```
## Your Environment

* OpenCAS version (commit hash or tag): 21.06.1.0547.release
* Operating System: Ubuntu 18.04.4 LTS
* Kernel version: `Linux node3 5.4.0-81-generic #91~18.04.1-Ubuntu SMP Fri Jul 23 13:36:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
* Cache device type (NAND/Optane/other): nvme
* Core device type (HDD/SSD/other): HDD
* Cache configuration:
* * Cache mode: (default: wt) wb
* * Cache line size: (default: 4)
* * Promotion policy: (default: always)
* * Cleaning policy: (default: alru)
* * Sequential cutoff policy: (default: full) always
* Other (e.g. _lsblk_, _casadm -P_, _casadm -L_)

```
casadm -L
type    id   disk        status    write policy   device
cache   1    /dev/dm-0   Running   wb             -
└core   1    /dev/dm-1   Active    -              /dev/cas1-1
```

**Comment by arutk** [comment]
Hi @inspur-wyq . We'll release a hotfix for 21.6 with patch for this issue.  Probably by the end of this week or early next week. Thank you for reporting this.

**Comment by arutk** [comment]
Fixed in release v21.6.3
